encoding.h

```diff
@@ -166,11 +167,19 @@
 #define PTE_A     0x040 // Accessed
 #define PTE_D     0x080 // Dirty
 #define PTE_SOFT  0x300 // Reserved for Software
-
+#define PTE_PPN 0x3ffffffffffc00
 #define PTE_PPN_SHIFT 10
+#define PTE_PPN_OFFST 2
```

Comment:
I think most of these constants can already be found in the header files

Reply:
OK. I will look into it to find some replacement